src/k8s-engine/lima.ts

```diff
@@ -953,7 +995,7 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
         await this.progressTracker.action(
           'Waiting for nodes',
           100,
-          this.client?.waitForReadyNodes() ?? Promise.reject(new Error('No client')));
+          this.client?.waitForReadyNodes() ?? await Promise.reject(new Error('No client')));
```

Comment:
That doesn't make sense: we're creating a new promise rejection, then awaiting for it to resolve? Also, doesn't `progressTracker.action` take a `Promsie<void> | () => Promise<void>`?

We probably want something like:
```suggestion
          async () => { if (!await this.client?.waitForReadyNodes()) { throw new Error('No client'); });
```

Reply:
So my IDE was complaining about the missing `await`, but I don't think the previous form makes sense. I'll pull the `await`s out, but could you write up a tech-debt issue on what exactly is going on here?  Either with or without the `await`, how does the stuff after the `??` ever fire?